api/scholarqa/scholar_qa.py

```diff
@@ -19,11 +21,15 @@
 from scholarqa.rag.retrieval import PaperFinder
 from scholarqa.state_mgmt.local_state_mgr import AbsStateMgrClient, LocalStateMgrClient
 from scholarqa.trace.event_traces import EventTrace
-from scholarqa.utils import get_paper_metadata, NUMERIC_META_FIELDS, CATEGORICAL_META_FIELDS
-from langsmith import traceable
+from scholarqa.utils import get_paper_metadata, NUMERIC_META_FIELDS, CATEGORICAL_META_FIELDS, get_ref_author_str, \
+    make_int
 
 logger = logging.getLogger(__name__)
 
+# Regular expressions to fix weird formatting issues cause after citation linking in the evidences
+CLOSE_BRACKET_PATTERN = r'(?<![\[|,\s*\d])(\d+\])'  # (Doe et al., 2024)10] --> (Doe et al., 2024)[10]
+OPEN_BRACKET_PATTERN = r"(\[[\d+,]+),(?=[^\[]*$)"  # [8,9,(Doe et al., 2024) --> [8,9](Doe et al., 2024)
+
```

Comment:
Moved from multistep_qa_pipeline.py

Reply:
sure, plz send some examples my way when you come across the weird formatting again